github/provider.go

```diff
@@ -239,13 +249,52 @@ func providerConfigure(p *schema.Provider) schema.ConfigureFunc {
 			}
 
 			token = appToken
-		}
 
-		writeDelay := d.Get("write_delay_ms").(int)
-		if writeDelay <= 0 {
-			return nil, fmt.Errorf("write_delay_ms must be greater than 0ms")
+			if owner == "" {
+				ctx := context.Background()
+
+				// Authenticate using the App Installation `token`
+				ts := oauth2.StaticTokenSource(
+					&oauth2.Token{AccessToken: token},
+				)
+
+				client := oauth2.NewClient(ctx, ts)
+				c := Config{
+					Token:      token,
+					BaseURL:    baseURL,
+					Insecure:   insecure,
+					WriteDelay: time.Duration(writeDelay) * time.Millisecond,
```

Comment:
Seems reasonable to omit configuring this (and the refactoring to define `writeDelay` earlier) as we only perform a read request to resolve the owner's login.